src/server/pkg/obj/microsoft_client.go

```diff
@@ -34,72 +26,59 @@ var (
 )
 
 type microsoftClient struct {
-	blobClient storage.BlobStorageClient
-	container  string
+	container *storage.Container
 }
 
 func newMicrosoftClient(container string, accountName string, accountKey string) (*microsoftClient, error) {
-	client, err := storage.NewBasicClient(
-		accountName,
-		accountKey,
-	)
+	client, err := storage.NewBasicClient(accountName, accountKey)
 	if err != nil {
 		return nil, err
 	}
-
-	return &microsoftClient{
-		blobClient: client.GetBlobService(),
-		container:  container,
-	}, nil
+	blobSvc := client.GetBlobService()
+	return &microsoftClient{container: (&blobSvc).GetContainerReference(container)}, nil
```

Comment:
Wait so `client.GetBlobService` returns a non-pointer and you then need to take its address to call methods on it? That's pretty lame.

Reply:
Yeah, I don't understand the reasoning for this.